### PR TITLE
Add overlay design patterns

### PR DESCRIPTION
**Pull request by daileytj**
Fixes PXBLUE-1312 & PXBLUE-1314 .

#### Changes proposed in this Pull Request:
- Add bottom sheet design pattern example
- Add complex bottom sheet design pattern example

### CODE REVIEW

**Comment by jeffvg** [comment]
looks good, can the icon alignment be adjusted up a bit for both?

image

**Comment by jeffvg** [comment]
question for all, do we want to match the text & icon colors to be the same from other examples ?

image

image

**Comment by huayunh** [comment]
> question for all

yes

**Comment by joebochill** [comment]
> looks good, can the icon alignment be adjusted up a bit for both?

The alignment looks correct...they're supposed to be centered when used with a mix of avatar/icons and left aligned only if there are just basic icons.

File: src/app/pages/overlays/complex-bottom-sheet/bottom-sheet/bottom-sheet.scss [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+.mat-h3 {
+    margin-bottom: 0;
+}
+
+.active {
+    .pxb-hero-label,
+    &.pxb-hero .pxb-hero-primary-wrapper {
+        color: map-get($pxb-blue, 500);
+    }
```
[the remark this answers is not in the document]
Comment (joebochill) [reply src/app/pages/overlays/complex-bottom-sheet/bottom-sheet/bottom-sheet.scss:11]: The classes on the Hero component don't look correct...are we using the latest version of the component library?
image
pxb-hero should be on the root tag, not the div, which is why the active style isn't applying correctly.

**Review by joebochill — COMMENTED** [review]
Bottom sheets out of the safe area (like we did for multiselect):
image

Code looks good otherwise

**Review by joebochill — APPROVED** [review]

**Pull request merged**